forest/cli/src/cli/info_cmd.rs

```diff
@@ -0,0 +1,404 @@
+    /// but averaging at 5 is considered healthy.
+    pub health: f64,
+    /// epoch the node is currently at
+    pub epoch: ChainEpoch,
+    /// BaseFee is the set price per unit of gas (measured in attoFIL/gas unit) to be burned (sent to an unrecoverable address) for every message execution
+    pub base_fee: TokenAmount,
+    /// sync status information
+    pub sync_status: SyncStatus,
+    /// Start time of the node
+    pub start_time: DateTime<Utc>,
+    /// Current network the node is running on
+    pub network: String,
+    /// Default wallet address selected.
+    pub default_wallet_address: Option<String>,
+    /// Default wallet address balance
+    pub default_wallet_address_balance: Option<String>,
+}
+
+pub struct NodeInfoOutput {
+    pub chain_status: ColoredString,
+    pub network: ColoredString,
+    pub uptime: ColoredString,
+    pub health: ColoredString,
+    pub wallet_address: ColoredString,
+    pub wallet_balance: ColoredString,
+}
+
+#[derive(Debug, strum::Display, PartialEq)]
+pub enum SyncStatus {
+    Ok,
+    Slow,
+    Behind,
+}
+
+// Limits the format string from humantime::format_duration to exclude us ns ms parts.
+pub fn fmt_duration(duration: Duration) -> String {
+    let duration = humantime::format_duration(duration);
+    let duration = duration.to_string();
+    let duration = duration.split(' ');
+    let format_duration = duration
+        .filter(|s| !s.ends_with("us"))
+        .filter(|s| !s.ends_with("ns"))
+        .filter(|s| !s.ends_with("ms"))
+        .map(|s| s.to_string());
+    let format_duration: Vec<String> = format_duration.collect();
+    format_duration.join(" ")
+}
+
+impl NodeStatusInfo {
+    pub fn new(
+        cur_duration: Duration,
+        blocks_per_tipset_last_finality: f64,
+        head: TipsetJson,
+    ) -> anyhow::Result<NodeStatusInfo> {
+        let ts = head.0.min_timestamp();
+        let cur_duration_secs = cur_duration.as_secs();
+        let behind = if ts <= cur_duration_secs + 1 {
+            cur_duration_secs.saturating_sub(ts)
+        } else {
+            anyhow::bail!(
+                "System time should not be behind tipset timestamp, please sync the system clock."
+            );
+        };
+
+        let sync_status = if behind < EPOCH_DURATION_SECONDS as u64 * 3 / 2 {
+            // within 1.5 epochs
+            SyncStatus::Ok
+        } else if behind < EPOCH_DURATION_SECONDS as u64 * 5 {
+            // within 5 epochs
+            SyncStatus::Slow
+        } else {
+            SyncStatus::Behind
+        };
+
+        let base_fee = head.0.min_ticket_block().parent_base_fee().clone();
+
+        let health = 100. * blocks_per_tipset_last_finality / BLOCKS_PER_EPOCH as f64;
+
+        Ok(Self {
+            behind: Duration::from_secs(behind),
+            health,
+            epoch: head.0.epoch(),
+            base_fee,
+            sync_status,
+            start_time: Utc::now(),
+            network: String::from("unknown"),
```

Comment:
This seems wrong - why even have the field if it's only ever set to `unknown`?

Reply:
It's an `Option<..>` now.